File: bitget-php-sdk-api/src/api/spot/SpotPlanApi.php
<?php


namespace bitget\api\spot;


use bitget\internal\BitgetApiClient;
use bitget\model\spot\plan\SpotPlanReq;
use bitget\model\spot\plan\SpotModifyPlanReq;
use bitget\model\spot\plan\SpotCancelPlanReq;
use bitget\model\spot\plan\SpotQueryPlanReq;

class SpotPlanApi
{
    const BASE_URL = "/api/spot/v1/plan";

    var BitgetApiClient $BitgetApiClient;

    public function __construct($BitgetApiClient)
    {
        $this->BitgetApiClient = $BitgetApiClient;
    }

    public function placePlan(SpotPlanReq $req): string
    {
        return $this->BitgetApiClient->doPost(self::BASE_URL . "/placePlan", $req);
    }

    public function modifyPlan(SpotModifyPlanReq $req): string
    {
        return $this->BitgetApiClient->doPost(self::BASE_URL . "/modifyPlan", $req);
    }

    public function cancelPlan(SpotCancelPlanReq $req): string
    {
        return $this->BitgetApiClient->doPost(self::BASE_URL . "/cancelPlan", $req);
    }

    public function currentPlan(SpotQueryPlanReq $req): string
    {
        return $this->BitgetApiClient->doPost(self::BASE_URL . "/currentPlan", $req);
    }

    public function historyPlan(SpotQueryPlanReq $req): string
    {
        return $this->BitgetApiClient->doPost(self::BASE_URL . "/historyPlan", $req);
    }

}

File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/spot/SpotPlanApi.java
package com.bitget.openapi.api.spot;

import com.bitget.openapi.dto.request.spot.*;
import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface SpotPlanApi {

    @POST("/api/spot/v1/plan/placePlan")
    Call<ResponseResult> placePlan(@Body SpotPlanOrderReq body);

    @POST("/api/spot/v1/plan/modifyPlan")
    Call<ResponseResult> modifyPlan(@Body SpotModifyPlanReq body);

    @POST("/api/spot/v1/plan/cancelPlan")
    Call<ResponseResult> cancelPlan(@Body SpotCancelPlanOrderReq body);

    @POST("/api/spot/v1/plan/currentPlan")
    Call<ResponseResult> currentPlan(@Body SpotPlanOpenOrderReq body);

    @POST("/api/spot/v1/plan/historyPlan")
    Call<ResponseResult> historyPlan(@Body SpotPlanOpenOrderReq body);
}